### PR TITLE
Corrected code for the 3 point energy correlator - Ananya Rai, Yale

### PR DESCRIPTION
**Pull request by anabananana98**

### CODE REVIEW

File: PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskJetsEEC.cxx [hunk 1]
```diff
@@ -394,14 +394,70 @@ void AliAnalysisTaskJetsEEC::ComputeEEC(AliEmcalJet *fJet, AliJetContainer *fJet
...
+//Filling the vectors
+                delta_Rvec.push_back(delta_R_js_2);
+                energy_pairs_vec.push_back(ee_js_2);
+                EEC_hist->Fill(delta_R_js_2,ee_js_2);
+                EEC_pt_hist->Fill(delta_R_js_2,jet_pt,ee_js_2);
+                EEC_pt_hist_log->Fill(log_delta_R_js_2,jet_pt,ee_js_2);
+////                fTreeEEC->Fill();
+                }//close s loop for the 2 point correlator
+            }//close if loop
+        }//close j loop
```
Comment (raymondEhlers) [inline PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskJetsEEC.cxx:434]: Careful here! The code may work as expected, but the indentation is mismatched, which makes it really easy to introduce bugs

File: PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskJetsEEC.cxx [hunk 2]
```diff
@@ -394,14 +394,70 @@ void AliAnalysisTaskJetsEEC::ComputeEEC(AliEmcalJet *fJet, AliJetContainer *fJet
 //Looping over the jet
     double jet_pt = fJet->Pt();
     jet_pt_hist->Fill(jet_pt); //filling histogram with momentum of jets
-//    constit = fConstitutents.constituents(); //vector of single jet constituents. Using pseudojet jet.constituent
+//For jets with 2 constituents
     for(int j=0; j<int(fConstituents.size()); j++)  //looping over constituents of the fConstituents object
         {
+         if(int(fConstituents.size()) == 2)
```
Comment (raymondEhlers) [inline PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskJetsEEC.cxx:400]: To make the logic a bit simpler and more efficient, you can swap the if statement with the outer for loop (and then you can move the next block where you deal with more than 2 constituents into the else block)

File: PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskJetsEEC.cxx [hunk 3]
```diff
@@ -449,8 +506,8 @@ void AliAnalysisTaskJetsEEC::ComputeEEC(AliEmcalJet *fJet, AliJetContainer *fJet
                 EEC_pt_hist->Fill(delta_R_js,jet_pt,ee_js);
                 EEC_pt_hist_log->Fill(log_delta_R_js,jet_pt,ee_js);
 ////                fTreeEEC->Fill();
-                }//close s loop
-             } //close j loop
+                }//close s loop for the 2 point correlator
+        } //close j loop
```
Comment (raymondEhlers) [inline PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskJetsEEC.cxx:510]: Appears to have the same mismatch

**Review by raymondEhlers — COMMENTED** [review]
Hi Ananya,

I left a few comments for you to consider. It can be approved as is, so I'm started the tests, but I would encourage you to consider implementing them - they will almost certainly save you some issues in the future

Best, Raymond

**Comment by raymondEhlers** [comment]
+test

File: PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskJetsEEC.cxx (hunk 1 above)
Comment (anabananana98) [reply PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskJetsEEC.cxx:434]: Hi Raymond, 

Thank you for your comments! I'll resolve them and then re-commit with the changes later today.

Best,
Ananya

**Pull request merged**